src/test/java/uk/gov/pay/api/utils/mocks/ConnectorDDMockClient.java

```diff
@@ -102,6 +105,21 @@ public void respondOk_whenGetMandateRequest(DDConnectorResponseToGetMandateParam
                         .withHeader(CONTENT_TYPE, APPLICATION_JSON)
                         .withBody(buildGetMandateResponse(params)));
     }
+    
+    public void respondOk_whenSearchMandatesRequest(Map<String, StringValuePattern> searchParams, SearchMandateConnectorResponse response, String gatewayAccountId) {
+        try {
+            String responseAsJson = new ObjectMapper().writeValueAsString(response);
+            wireMockClassRule.stubFor(get(urlPathEqualTo(format(CONNECTOR_MOCK_MANDATES_PATH, gatewayAccountId)))
+                    .withHeader(ACCEPT, matching(APPLICATION_JSON))
+                    .withQueryParams(searchParams)
+                    .willReturn(aResponse()
+                    .withStatus(200)
+                    .withHeader(CONTENT_TYPE, APPLICATION_JSON)
+                    .withBody(responseAsJson)));
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
```

Comment:
Since this is only used in tests, is there any reason why we can’t just have this exception bubble up all the way to the top?

Reply:
I thought since it was test then no harm in catching an throwing it to save noise on the method signatures, but if you think cleaner to pass it up then so be it. I've changed 👍